sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -139,48 +139,47 @@ object MetalsPlugin extends AutoPlugin {
         javaHome: File
     ): Option[JdkVersion] = {
 
-      def fromReleaseFile: Option[JdkVersion] = {
-        val releaseFile = javaHome / "release"
-        if (releaseFile.exists) {
-          val props = new ju.Properties
-          props.load(Source.fromFile(releaseFile).bufferedReader())
-          try {
+      def fromReleaseFile: Option[JdkVersion] =
+        Seq(javaHome / "release", javaHome.getParentFile / "release")
+          .filter(_.exists)
+          .flatMap { releaseFile =>
+            val props = new ju.Properties
+            props.load(Source.fromFile(releaseFile).bufferedReader())
             props.asScala
               .get("JAVA_VERSION")
               .map(_.stripPrefix("\"").stripSuffix("\""))
-              .flatMap(JdkVersion.parse)
-          } catch {
-            case NonFatal(e) =>
-              None
+              .flatMap(jv => JdkVersion.parse(jv))
           }
-        } else None
-      }
+          .headOption
 
-      def jdk8Fallback: Option[JdkVersion] = {
-        val rtJar = javaHome / "jre" / "lib" / "rt.jar"
-        if (rtJar.exists) Some(JdkVersion(8))
-        else None
-      }
+      def jdk8Fallback: Option[JdkVersion] =
+        Seq(javaHome / "jre" / "lib" / "rt.jar", javaHome / "lib" / "rt.jar")
+          .filter(_.exists)
+          .map(_ => JdkVersion(8))
+          .headOption
 
       fromReleaseFile.orElse(jdk8Fallback)
     }
 
     def parse(v: String): Option[JdkVersion] = {
```

Comment:
Take it or leave it, but do you think this is easier to read:

```scala
    def parse(v: String): Option[JdkVersion] = {
      val numbers = Try {
        v
          .split('-')
          .head
          .split('.')
          .toList
          .take(2)
          .flatMap(s => Try(s.toInt).toOption)
      }.toOption

      numbers match {
        case Some(1 :: minor :: _) =>
          Some(JdkVersion(minor))
        case Some(single :: _) =>
          Some(JdkVersion(single))
        case _ => None
      }
    }
```
Might simplify it a bit.

Reply:
:+1:

Reply:
btw, I also realized that there is no need in first split by `-`
```
      val numbers = Try {
        v
          .split('.')
          .toList
          .take(2)
          .flatMap(s => Try(s.toInt).toOption)
      }.toOption
```

Reply:
The split by `-` is actually needed to parse the Java version `17-ea`, as in this [test](https://github.com/scalameta/metals/blob/3a9ab1d71c3275276e1e0e44f17264fc33d0bfaf/tests/unit/src/test/scala/tests/JavaInteractiveSemanticdbSuite.scala#L10).

Reply:
oh, yep. sorry for the wrong suggestion!